base/range.jl

```diff
@@ -31,8 +31,9 @@ _colon(::Any, ::Any, start::T, step, stop::T) where {T} =
 """
     (:)(start, [step], stop)
 
-Range operator. `a:b` constructs a range from `a` to `b` with a step size of 1 (a [`UnitRange`](@ref))
-, and `a:s:b` is similar but uses a step size of `s` (a [`StepRange`](@ref)).
+Range operator. `a:b` constructs a range from `a` to `b` with an implicit step size
+of 1 (often a [`UnitRange`](@ref)), and `a:s:b` is similar but uses an explicit step
+size of `s` (a [`StepRange`](@ref) or [`StepRangeLen`](@ref)).
 
 `:` is also used in indexing to select whole dimensions, e.g. in `A[:, 1]`.
```

Comment:
```suggestion
The function `:` is also used in indexing to select whole dimensions, e.g. in `A[:, 1]`.

The syntax `:` is also used to [`quote`](@ref) code, e.g. `:(x + y) isa Expr` and `:x isa Symbol`.
Since `:2 isa Int`, it does *not* create a range in indexing: `v[:2] == v[2] != v[begin:2]`.
```

Reply:
This part is great, but I feel its not "compat", like there should be no addition of words like "function" here, since one can then argue also for adding "operators" to the others.

I feel as the current docs is, the two new useful cases you introduced should just be added to the next line, like:
```julia
Range operator. a:b constructs a range from a to b with a step size of 1 (a
UnitRange) , and a:s:b is similar but uses a step size of s (a StepRange).

: is used in indexing to select whole dimensions and for Symbol literals, as
in e.g. :hello.

: is also used to quote code, e.g. :(x + y) isa Expr and :x isa Symbol.
Since :2 isa Int, it does not create a range in indexing: v[:2] == v[2] != v[begin:2].
```